src/set-project-version.sh

```diff
@@ -33,6 +33,9 @@ pushd $(dirname "$0")/..
 mvn versions:set -DnewVersion=$NEW_VERSION
 mvn versions:set -DnewVersion=$NEW_VERSION -pl buildtools
 mvn versions:set -DnewVersion=$NEW_VERSION -pl pulsar-sql/presto-distribution
+# install the new version of root pom local, so `update-parent` can update the right parent version
+mvn -am -pl . clean install -DskipTests
+mvn versions:update-parent -DallowSnapshots=true -DparentVersion=$NEW_VERSION -f protobuf-shaded/pom.xml
 mvn versions:set -DnewVersion=$NEW_VERSION -f protobuf-shaded/pom.xml
```

Comment:
If update-parent worked well, then we could remove this line?  @merlimat

Reply:
no. we need both.

update-parent is to update the parent version; `set` is to set the version.